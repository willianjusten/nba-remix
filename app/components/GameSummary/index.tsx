import { OvertimeHead, OvertimeScore } from '~/components/Overtime'
import { Table, TableCell, TableHead } from '~/components/Table'
import { TeamScore } from '~/types'

export type GameSummaryProps = {
  game: {
    period: number
    hTeam: TeamScore
    vTeam: TeamScore
  }
}

function GameSummary({ game }: GameSummaryProps) {
  return (
    <div className="py-5">
      <h1 className="text-2xl font-semibold">Game Summary</h1>
      <div className="overflow-x-auto">
        <Table>
          <TableHead>
            <tr className="border border-slate-600">
              <TableCell>Team</TableCell>
              <TableCell>Q1</TableCell>
              <TableCell>Q2</TableCell>
              <TableCell>Q3</TableCell>
              <TableCell>Q4</TableCell>
              <OvertimeHead period={game.period} />
            </tr>
          </TableHead>
          <tbody>
            <tr>
              <TableCell>{game.hTeam.ta}</TableCell>
              <TableCell>{game.hTeam.q1}</TableCell>
              <TableCell>{game.hTeam.q2}</TableCell>
              <TableCell>{game.hTeam.q3}</TableCell>
              <TableCell>{game.hTeam.q4}</TableCell>
              <OvertimeScore period={game.period} team={game.hTeam} />
            </tr>
            <tr>
              <TableCell>{game.vTeam.ta}</TableCell>
              <TableCell>{game.vTeam.q1}</TableCell>
              <TableCell>{game.vTeam.q2}</TableCell>
              <TableCell>{game.vTeam.q3}</TableCell>
              <TableCell>{game.vTeam.q4}</TableCell>
              <OvertimeScore period={game.period} team={game.vTeam} />
            </tr>
          </tbody>
        </Table>
      </div>
    </div>
  )
}

export default GameSummary
