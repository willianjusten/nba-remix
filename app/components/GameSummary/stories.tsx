import { Story, Meta } from '@storybook/react'
import GameSummary, { GameSummaryProps } from '.'

export default {
  title: 'GameSummary',
  component: GameSummary,
  parameters: {
    backgrounds: {
      default: 'dark',
    },
  },
} as Meta

export const Default: Story<GameSummaryProps> = (args) => (
  <GameSummary {...args} />
)

Default.args = {
  game: {
    period: 6,
    hTeam: {
      ta: 'Team A',
      q1: '10',
      q2: '20',
      q3: '30',
      q4: '40',
      ot1: '12',
      ot2: '13',
    },
    vTeam: {
      ta: 'Team B',
      q1: '10',
      q2: '20',
      q3: '30',
      q4: '40',
      ot1: '12',
      ot2: '13',
    },
  },
}
