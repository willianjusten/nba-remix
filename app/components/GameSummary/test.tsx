import { render } from '@testing-library/react'
import GameSummary from '.'

describe('<GameSummary>', () => {
  it('should render correctly', () => {
    const game = {
      period: 6,
      hTeam: {
        ta: 'Team A',
        q1: '10',
        q2: '20',
        q3: '30',
        q4: '40',
        ot1: '12',
        ot2: '13',
      },
      vTeam: {
        ta: 'Team B',
        q1: '10',
        q2: '20',
        q3: '30',
        q4: '40',
        ot1: '12',
        ot2: '13',
      },
    }

    const { container } = render(<GameSummary game={game} />)

    // in cases where its a big content
    // its better to use the snapshot testing
    expect(container).toMatchSnapshot()
  })
})
