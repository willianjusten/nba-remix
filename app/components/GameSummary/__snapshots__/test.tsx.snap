// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameSummary> should render correctly 1`] = `
<div>
  <div
    class="py-5"
  >
    <h1
      class="text-2xl font-semibold"
    >
      Game Summary
    </h1>
    <div
      class="overflow-x-auto"
    >
      <table
        class="my-5 min-w-full border border-slate-600 bg-glass text-center text-white
      md:min-w-min
    "
      >
        <thead
          class="bg-slate-900 font-bold"
        >
          <tr
            class="border border-slate-600"
          >
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              Team
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              Q1
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              Q2
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              Q3
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              Q4
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              OT
              1
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              OT
              2
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              Team A
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              10
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              20
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              30
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              40
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              12
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              13
            </td>
          </tr>
          <tr>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              Team B
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              10
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              20
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              30
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              40
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              12
            </td>
            <td
              class="border border-slate-500 px-3 py-2 "
            >
              13
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
